pinot-perf/src/main/java/org/apache/pinot/perf/BenchmarkStringVarLengthDictionary.java

```diff
@@ -0,0 +1,198 @@
+    while (i < DICTIONARY_LENGTH) {
+      String randomString = RandomStringUtils.randomAlphanumeric(
+          USE_FIXED_SIZE_STRING ? MAX_STRING_LENGTH : (1 + random.nextInt(MAX_STRING_LENGTH)));
+      if (uniqueStrings.contains(randomString)) {
+        continue;
+      } else {
+        uniqueStrings.add(randomString);
+      }
+      _inputData[i] = randomString;
+      GenericRow genericRow = new GenericRow();
+      genericRow.putValue(COLUMN_NAME, _inputData[i++]);
+      rows.add(genericRow);
+    }
+
+    // Generate Segment with fixed dictionary
+    SegmentGeneratorConfig fixedDictionarySegmentGeneratorConfig = new SegmentGeneratorConfig(tableConfig, schema);
+    fixedDictionarySegmentGeneratorConfig.setOutDir(fixedLengthIndexDir.getParent());
+    fixedDictionarySegmentGeneratorConfig.setFormat(FileFormat.AVRO);
+    fixedDictionarySegmentGeneratorConfig.setSegmentName(segmentName);
+
+    SegmentIndexCreationDriverImpl driver = new SegmentIndexCreationDriverImpl();
+    driver.init(fixedDictionarySegmentGeneratorConfig, new GenericRowRecordReader(rows));
+    driver.build();
+
+    // Update table config for variable length string dictionary
+    ArrayList<String> varLengthDictionaryColumns = new ArrayList<>();
+    varLengthDictionaryColumns.add(COLUMN_NAME);
+    tableConfig.getIndexingConfig().setVarLengthDictionaryColumns(varLengthDictionaryColumns);
+
+    // Generate Segment with var length dictionary
+    SegmentGeneratorConfig varLengthDictionarySegmentGeneratorConfig = new SegmentGeneratorConfig(tableConfig, schema);
+    varLengthDictionarySegmentGeneratorConfig.setOutDir(varLengthIndexDir.getParent());
+    varLengthDictionarySegmentGeneratorConfig.setFormat(FileFormat.AVRO);
+    varLengthDictionarySegmentGeneratorConfig.setSegmentName(varSegmentName);
+    driver = new SegmentIndexCreationDriverImpl();
+    driver.init(varLengthDictionarySegmentGeneratorConfig, new GenericRowRecordReader(rows));
+    driver.build();
+
+    _immutableSegmentWithFixedDictionary = ImmutableSegmentLoader.load(fixedLengthIndexDir, ReadMode.mmap);
+    _immutableSegmentWithVarLengthDictionary = ImmutableSegmentLoader.load(varLengthIndexDir, ReadMode.mmap);
+  }
+
+  @TearDown
+  public void tearDown()
+      throws Exception {
+    _immutableSegmentWithFixedDictionary.destroy();
+    _immutableSegmentWithVarLengthDictionary.destroy();
+    FileUtils.deleteDirectory(TMP_DIR);
+  }
+
+  @Benchmark
+  public void fixedStringDictionaryIndexOf() {
+    Dictionary fixedLengthStringDictionary = _immutableSegmentWithFixedDictionary.getDictionary(COLUMN_NAME);
+    for (Integer i : _randomReadOrder) {
+      fixedLengthStringDictionary.indexOf(_inputData[i]);
+    }
+  }
+
+  @Benchmark
+  public void fixedStringDictionaryGet() {
+    Dictionary fixedLengthStringDictionary = _immutableSegmentWithFixedDictionary.getDictionary(COLUMN_NAME);
+    for (Integer i : _randomReadOrder) {
+      fixedLengthStringDictionary.get(i);
+    }
+  }
```

Comment:
You need to pass a `Blackhole bh` parameter to the benchmark and replace`fixedLengthStringDictionary.get(i);` with `bh.consume(fixedLengthStringDictionary.get(i));` to get reliable results. Otherwise the benchmark may get partially or entirely eliminated. There's also no need to box `I` unless I'm missing something?

Reply:
Hi @richardstartin, Thank you for the suggestions. I fixed the test using `bh.consume()`. For your comment regarding `no need to box I`, I guess that you were referring to `for (Integer i: _randomReadOrder)`? I changed that part to use the for loop with the integer value.. Also, I made a mistake on the above function `dictionary.get(i) -> dictionary.get(_randomOrder[i])`

Reply:
Yes i mean change `Integer` to `int` so there's less noise